Automattic/woocommerce-payments
# fix: WooPay 'Save my info' spacing

#### Changes proposed in this Pull Request









I noticed that WooPay's "Save my info" addins unnecessary spacing (via margins) to the checkout pages when no content is displayed.
As good citizens of the checkout pages, fixing to ensure consistent behavior compared to when WooPay is disabled or not available.

Cannot be tested via UTs.

Before/After - WooPay disabled - spacing is consistent:
![Screenshot 2023-10-06 at 12 19 30 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/da3057b4-dc5f-4a34-8a1d-aa103476ec21)


Before - no payment method needed - "Save my info" is adding a margin-top:
![Screenshot 2023-10-06 at 12 24 47 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/67ba6f34-ece1-43fe-bf55-9cb6a6ee1e15)


Before - non-CC payment method is selected (or existing WooPay user is being used) - "Save my info" is adding a margin-top:
![Screenshot 2023-10-04 at 3 41 03 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/df1018f3-cd7e-40df-a046-a6292d849a39)

After - non-CC payment method is selected - "Save my info" is no longer adding a margin-top:
![Screenshot 2023-10-04 at 3 39 07 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/b6aec958-41cf-4d6d-8db7-a52cc60c8f6a)

Before/After - CC Payment method is selected:
![Screenshot 2023-10-04 at 3 39 19 PM](https://github.com/Automattic/woocommerce-payments/assets/273592/e79ca64b-0133-4f7e-aaac-6c1bcf33a17c)


#### Testing instructions





- Enable WooPay in the WC Payments settings
- Enable other non-WooPayments payment methods (e.g.: cash on delivery)
- Add a simple product to the cart, go to checkout
- Select CC as a payment method
- Enter a valid WooPay email
  - "Save my info" no longer displayed, notice spacing above "pay now" and below the payment methods
- Enter a non-existing WooPay email
  - "Save my info" is displayed, notice the spacing
- Select a non-CC payment method (e.g.: cash on delivery)
  - "Save my info" no longer displayed and no longer adding a margin-top

-------------------


**Post merge**
